r-package/grf/R/causal_forest.R

```diff
@@ -133,6 +133,8 @@ causal_forest <- function(X, Y, W,
                           num.fit.reps = 50,
                           num.optimize.reps = 1000) {
     validate_X(X)
+    validate_Y(Y)
+    validate_W(W)
     if(length(Y) != nrow(X)) { stop("Y has incorrect length.") }
```

Comment:
It would be nice to move these checks into the validate methods as well, to unify all the validation logic in one place.

Reply:
@jtibshirani Thanks for the suggestion.  Did you mean move the `if(length(Y))...` and `if(length(W)...` tests into the validate_Y/W functions?  That would require passing X, or its length, to the validate_Y/W functions.  Is that ok?  Another alternative would be to create a wrapper for these 5 lines as `validate_XYW(X,Y,W)`.  Let me know which you would prefer.  Thanks.

Reply:
I've modified this to use `validate_YX`, `validate_WX` which then call `validate_Y` and `validate_W` to keep the logic separate.  I also added a `validate_ZX` and `validate_Z` for the instrumental_forest.  This way it offers both YX and Y-only validation.